stellargraph/mapper/adjacency_generators.py

```diff
@@ -0,0 +1,84 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+import numpy as np
+from ..core import StellarGraph
+from ..core.utils import normalize_adj
+from ..core.tf_utils import partial_powers, select_row_from_sparse_tensor
+from ..core.experimental import experimental
+
+
+@experimental(reason="lack of unit tests")
+class AdjacencyPowerGenerator:
+    """
+    A data generator for use with the Watch Your Step algorithm [1]. It calculates and returns the first `num_powers`
+    of the adjacency matrix row by row.
+
+    .. warning::
+
+        This class is experimental: it is insufficiently tested.
+
+    Args:
+        G (StellarGraph): a machine-learning StellarGraph-type graph
+        num_powers (int): the number of adjacency powers to calculate
+
+    """
+
+    def __init__(self, G, num_powers=5):
+
+        if not isinstance(G, StellarGraph):
+            raise TypeError("G must be a StellarGraph object.")
+
+        Aadj = G.to_adjacency_matrix()
+        indices = list(zip(Aadj.col, Aadj.row))
+
+        self.Aadj_T = tf.sparse.SparseTensor(
+            indices=indices,
+            values=Aadj.data.astype(np.float32),
+            dense_shape=(Aadj.shape[0], Aadj.shape[0]),
```

Comment:
I think:

```suggestion
            dense_shape=Aadj.shape,
```

- [ ] and similarly in `self.transition_matrix_T`

Reply:
yep, neater